jupyterlab/extensions/pypi.py

```diff
@@ -13,14 +14,17 @@
 from datetime import datetime, timedelta, timezone
 from functools import partial
 from itertools import groupby
+from os import environ
 from pathlib import Path
 from subprocess import CalledProcessError, run
 from tarfile import TarFile
 from typing import Any, Callable, Dict, List, Optional, Tuple
+from urllib.parse import urlparse
 from zipfile import ZipFile
 
 import tornado
 from async_lru import alru_cache
+from tornado.curl_httpclient import CurlAsyncHTTPClient
```

Comment:
Hmm it depends on `pycurl` and a specific build and version of libcurl is also required as per [docs](https://www.tornadoweb.org/en/stable/httpclient.html#module-tornado.httpclient).

Not sure if we can just import it conditionally (if proxy is enabled, otherwise use `AsyncHTTPClient`) or if we prefer to add it as a dependency anyways. What others think?

Reply:
That dependency is likely to be annoying (looking at you Windows...). I would prefer adding a new dependency like [aiohttp](https://github.com/aio-libs/aiohttp) or [httpx](https://www.python-httpx.org/) (I checked they support proxy).

@dolevf would you mind changing this PR for one of the above ?

Reply:
... and https://github.com/curl/curl/discussions/12026, so not ideal timing.

Reply:
Before considering fresh dependencies (and their dependencies), which _inevitably_ bring fresh headaches, perhaps a route would be to consider hoisting to a configurable, defaulting to the current client, and making the `libcurl`-based one (which _is_ used on e.g. `nbviewer`). Someone would then be able to relatively easily overload this behavior in an extension, which would make it feasible to enable-on-install with the `{sys.prefix}/etc/jupyter/jupyter_server_config.d/` deployment approach.

e.g. 

```yaml
{
  "PyPIExtensionManager": {
    "pypi_client_cls": "jupyterlab.extensions.pypi.clients.TornadoPyPI",
    # vs                jupyterlab.extensions.pypi.clients.TornadoCurlPyPI
    # vs                jupyterlab_httpx.pypi.HttpxPyPI
  }
}
```

Reply:
I would be in favor of using `httpx` which is very lightweight and somewhat more modern than `aiohttp`, which also comes with a server.

Reply:
Hi,

I've changed it to `httpx`. I've never worked with Jupyterlab code base before, so I couldn't find a great way to test it end to end other than ensuring it works locally against a squid proxy.